modules/twinklexfd.js

```diff
@@ -1067,49 +1067,64 @@ Twinkle.xfd.callbacks = {
-					break;
-				case 'tfm':
-					notifytext += '{{subst:Tfm notice|1=' + mw.config.get('wgTitle') + '|2=' + params.target + modNotice + '}} ~~~~';
-					break;
-				default:
-					alert('twinklexfd in userNotification: unknown TFD action');
-					break;
+			// Notify developer(s) of script(s) that use(s) the nominated template
+			if (inCategories.includes('Templates used by Twinkle')) {
+				pagesToNotify.push('Wikipedia talk:Twinkle');
+			}
+			if (inCategories.includes('Templates used by AutoWikiBrowser')) {
+				pagesToNotify.push('Wikipedia talk:AutoWikiBrowser');
+			}
+			if (inCategories.includes('Templates used by RedWarn')) {
+				pagesToNotify.push('Wikipedia talk:RedWarn');
 			}
 
-			usertalkpage.setAppendText(notifytext);
-			usertalkpage.setEditSummary('Notification: [[' + params.discussionpage + '|listing]] of [[:' + pageobj.getPageName() + ']] at [[WP:TFD|templates for discussion]].' + Twinkle.getPref('summaryAd'));
-			usertalkpage.setCreateOption('recreate');
-			Twinkle.xfd.setWatchPref(usertalkpage, Twinkle.getPref('xfdWatchUser'));
-			usertalkpage.setFollowRedirect(true);
+			pagesToNotify.forEach(function(pagename, i) {
+				var pageToNotify = new Morebits.wiki.page(pagename, 'Notifying of template nomination');
+				var notifytext = '\n';
+				var modNotice = mw.config.get('wgPageContentModel') === 'Scribunto' ? '|module=yes' : '';
+				switch (params.xfdcat) {
+					case 'tfd':
+						notifytext += '{{subst:Tfd notice|1=' + mw.config.get('wgTitle') + modNotice + '}} ~~~~';
+						break;
+					case 'tfm':
+						notifytext += '{{subst:Tfm notice|1=' + mw.config.get('wgTitle') + '|2=' + params.target + modNotice + '}} ~~~~';
+						break;
+					default:
+						alert('twinklexfd in courtesyNotification: unknown TFD action');
+						break;
+				}
 
-			// Add this nomination to user's userspace log, if the user has enabled it
-			// and it isn't the second template in a TfM nomination
-			if ((params.xfdcat === 'tfd' || pageobj.getPageName() === Morebits.pageNameNorm) && params.lognomination) {
-				usertalkpage.append(function onNotifySuccess() {
-					Twinkle.xfd.callbacks.addToLog(params, initialContrib);
-				}, function onNotifyError() {
-					// if user could not be notified, log without mentioning notification
-					Twinkle.xfd.callbacks.addToLog(params, null);
-				});
-			} else {
-				usertalkpage.append();
-			}
+				pageToNotify.setAppendText(notifytext);
+				pageToNotify.setEditSummary('Notification: [[' + params.discussionpage + '|listing]] of [[:' + pageobj.getPageName() + ']] at [[WP:TFD|templates for discussion]].' + Twinkle.getPref('summaryAd'));
+				pageToNotify.setCreateOption('recreate');
+				pageToNotify.setFollowRedirect(true);
+				Twinkle.xfd.setWatchPref(pageToNotify, Twinkle.getPref('xfdWatchUser'));
+
+				// Only add to log the first time around.
+				if (i === 1 && (params.xfdcat === 'tfd' || pageobj.getPageName() === Morebits.pageNameNorm) && params.lognomination) {
```

Comment:
I actually think this would be better using a separate function, rather than usurping the `userNotification`, for roughly this reason.  Something like what I did for redirect target notification in #527, albeit without the preference and box.  Clearer, less fudging, and allows for more specific edit summaries, etc.

Reply:
Changed to i === 0. Didn't make a new function, but I can if you think it's better. It definitely would be less hack-y, but also would be a bit more work :).

Reply:
I do, actually, sorry.  Happy to be proven wrong, but I think it'd be cleaner and easier to maintain going forward, especially around the notification->user log interface.

I'm largely unavailable for the next week or so, but I'd be happy to help you draft it up next week if you'd like!